### PR TITLE
[SELC-5314] feat: SFE Billing Code field validation

### PR DESCRIPTION
**Pull request by GabrieleMaiocchiFilippo**
#### List of Changes

[SELC-5314] feat: SFE Billing Code field validation

#### Motivation and Context

the field for sfe has to be free and be validated calling the API verifyTaxCodeInvoicing

#### How Has This Been Tested?

Tested if when landing in page the field SFE is free and editable.
Tested if at the change of the value SFE field the API verifyTaxCodeInvoicing has been called with the correct values and that has been retrived the correct response.

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

[SELC-5314]: https://pagopa.atlassian.net/browse/SELC-5314?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/views/onboardingProduct/__tests__/OnboardingProduct.test.tsx [hunk 1]
```diff
@@ -1198,8 +1210,9 @@ const billingData2billingDataRequest = () => ({
   digitalAddress: 'a@a.it',
   zipCode: '09010',
   taxCode: 'AAAAAA44D55F456K',
+  taxCodeInvoicing: 'AAAAAA44D55F456K',
```
Comment (loremedda) [inline src/views/onboardingProduct/__tests__/OnboardingProduct.test.tsx:1213]: ? if i check in the mocks, this taxCode is not present

File: src/views/onboardingProduct/__tests__/OnboardingProduct.test.tsx [hunk 2]
```diff
@@ -904,6 +910,12 @@ const fillUserBillingDataForm = async (
   fireEvent.change(document.getElementById(vatNumber) as HTMLElement, {
     target: { value: 'AAAAAA44D55F456K' },
   });
+
+  const taxCodeInvoicing = document.getElementById('taxCodeInvoicing');
+  if (taxCodeInvoicing) {
+    fireEvent.change(taxCodeInvoicing, { target: { value: 'AAAAAA44D55F456K' } });
```
Comment (loremedda) [inline src/views/onboardingProduct/__tests__/OnboardingProduct.test.tsx:916]: ? if i check in the mocks, this taxCode is not present

**Review by loremedda — CHANGES_REQUESTED** [review]

**Review by loremedda — CHANGES_REQUESTED** [review]
Unit tests must ensure that a functionality that has been tested with a behavior test that satisfies the business requirement..
From your code I see that the call is invoked when the 11th character is reached, here you would be inserting 16 characters
The user can potentially enter up to 100 characters as you have not set a length limit on the textfield

**Review by loremedda — CHANGES_REQUESTED** [review]
Add a positive use case in the mocks and test that when it is inserted, the validation is successful, otherwise the error we expect. Make sure the feature works locally first, then testing

**Review by loremedda — APPROVED** [review]

**Review by gianmarcoplutino — APPROVED** [review]

**Pull request merged**